### PR TITLE
[Feature]: Aborting jobs on synapse

### PR DESCRIPTION
**Pull request by sachinkumargit**
# Issue Link

https://github.com/LambdaTest/test-at-scale/issues/202

# Description

Added functionality to abort builds on synapse.

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] Tested using github push webhook for various stages of build like discovery, execution.

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: pkg/runner/docker/docker.go [hunk 1]
```diff
@@ -337,6 +338,19 @@ func (d *docker) KillRunningDocker(ctx context.Context) {
 	}
 }
 
+func (d *docker) KillContianerForBuildID(buildID string) error {
```
Comment (saurabh-prakash) [inline pkg/runner/docker/docker.go:341]: typo

File: pkg/synapse/synapse.go [hunk 2]
```diff
@@ -215,6 +216,9 @@ func (s *synapse) processMessage(msg []byte, duplicateConnectionChan chan struct
 	case core.MsgTask:
 		s.logger.Debugf("task message received from server")
 		go s.processTask(message)
+	case core.MsgBuildAbort:
+		s.logger.Debugf("abort-build message received from server")
```
Comment (VikrantKS) [inline pkg/synapse/synapse.go:220]: can we add more information here , like orgID and buildID

File: pkg/runner/docker/docker.go (hunk 1 above)
Comment (sachinkumargit) [reply pkg/runner/docker/docker.go]: fixed.

File: pkg/core/runner.go [hunk 3]
```diff
@@ -64,6 +64,9 @@ type DockerRunner interface {
 	// KillRunningDocker kills  container spawn by synapse
 	KillRunningDocker(ctx context.Context)
 
+	// KillContainerForBuildID kills synapse container which is running build for given buildID
```
Comment (VikrantKS) [inline pkg/core/runner.go:67]:
```suggestion
	// KillContainerForBuildID kills synapse container which is running for given buildID
```

**Review by VikrantKS — APPROVED** [review]

**Pull request merged**